tabbycat/standings/teams.py

```diff
@@ -183,7 +183,7 @@ def annotate(self, queryset, standings, round=None):
 
         # Make a copy of teams queryset and annotate with opponents
         opponents_filter = ~Q(debateteam__debate__debateteam__team_id=F('id'))
-        opponents_filter &= Q(debateteam__debate__round__stage=Round.Stage.PRELIMINARY)
+        opponents_filter &= Q(debateteam__debate__round__stage=Round.Stage.PRELIMINARY, debateteam__teamscore__ballot_submission__confirmed=True)
```

Comment:
I haven't looked at this for quite a long time, of course, but this surprises me a bit.

The original queryset is a `Team` queryset, right? But you're saying when a debate has (say) a confirmed and an unconfirmed ballot, it gets counted twice? But this calculation doesn't actually use the ballots, because `opponents_annotation` just returns an array of all the opponents? So why is it joining to the ballot submissions table in the first place?

Reply:
All the joins are made because the query is repeated from the combinable metrics' query, but you're right, it should be a clean `Team` query.

Doing this, I think there could be a couple further optimizations:
- Use `FilteredRelation` to avoid filtering annotations on the same things
- If the draw strength metric is combinable, include it in the main query to avoid this extra call.

As this would be for a hotfix, I'll think more about the optimizations later.